onnxruntime/core/optimizer/attention_fusion_helper.h

```diff
@@ -593,6 +610,52 @@ bool MatchInputMaskSubgraph(const Graph& graph, const Node& qkv_matmul, Attentio
   return true;
 }
 
+bool MatchInputMaskSubgraph(const Graph& graph, const Node& qkv_matmul, AttentionMaskNodesDistilBert& result, const logging::Logger& logger) {
+  DEBUG_LOG("Start MatchInputMaskSubgraphDistilBert");
+
+  std::vector<graph_utils::EdgeEndToMatch> mask_path{
+      {0, 0, "Softmax", {1, 11, 13}, kOnnxDomain},
+      {0, 0, "Where", {9}, kOnnxDomain},
+      {0, 0, "Expand", {8, 13}, kOnnxDomain},
+      {0, 0, "Reshape", {1, 5, 13}, kOnnxDomain},
+      {0, 0, "Equal", {1, 7, 11, 13}, kOnnxDomain}};
+
+  std::vector<const Node::EdgeEnd*> edges;
+  if (!graph_utils::FindPath(qkv_matmul, true, mask_path, edges, logger)) {
+    DEBUG_LOG("Failed to find mask path");
+    return false;
+  }
+
+  const Node& softmax = edges[0]->GetNode();
+  const Node& where = edges[1]->GetNode();
```

Comment:
Please check the following:
(1) Where has X=-Infinity
(2) Expand has another input Shape <-- qk_MatMul
(3) Equal has input B=0
(4) Reshape node's shape input